app/decorators/course_decorator.rb

```diff
@@ -505,6 +505,11 @@ def course_fee_content
     )
   end
 
+  def no_fee?
+    object.teacher_degree_apprenticeship? ||
+      (fee_international.blank? && fee_uk_eu.blank?)
```

Comment:
Courses have a funding column now, should that be used to determine if the course is fee based?

Reply:
@inulty-dfe do you mean the no_fee only checks for `funding == 'fee'` or something similar? cc @avinhurry

Reply:
@avinhurry what do you think about?

Reply:
What is the intention with this `no_fee` method? @tomas-stefano

Reply:
> I don't think we clear out the fee information when changing from a non TDA course to a TDA course? We should probably add a card to do that.

This PR does that (on tda attributes service class change)

Reply:
> Should TDA courses never have the fee information inputted?

TDA specifically yes. Other future undergraduate courses I am not sure.

Reply:
Hmm, i'd like to go with Iains suggestion of using `funding == fee` but i'm worried about edge cases. 

Why do we need this check? `object.teacher_degree_apprenticeship?`

Am I right in thinking that this check is only to catch the edge cases that are currently in? If so it might be better to fix those while their aren't that many. Maybe we should have a quick huddle.

Reply:
@avinhurry @inulty-dfe that's done ✌️